zzz

```diff
@@ -36,7 +36,10 @@ case "$ZZZ_MODE" in
   hibernate) grep -q disk /sys/power/state || fail "hibernate not supported";;
 esac
 
-test -w /sys/power/state || fail "sleep permission denied"
+if ! [ -w /sys/power/state ] || {
```

Comment:
! ... || is a double-negative and makes it highly painful to understand what the logic is supposed to be in this if statement. nested ifs would make this much more readable